feat: codemod Update virtual network and storage account modules

Kingshuk-Microsoft opened the pull request:
- Deleted the old virtual network module and replaced it with a new implementation.
- Updated storage account module to use a new SKU default and improved parameter handling.
- Enhanced the test cases for enterprise-grade configurations, including new parameters for Azure AI services.
- Bumped version to 0.2 to reflect the changes made across modules.

## Description

<!--
>Thank you for your contribution !
> Please include a summary of the change and which issue is fixed.
> Please also include the context.
> List any dependencies that are required for this change.

Fixes #123
Closes #123
-->

## Pipeline Reference

<!-- Insert your Pipeline Status Badge below -->

| Pipeline |
| -------- |
[![avm.ptn.sa.modernize-your-code](https://github.com/Ravikirana-Microsoft/bicep-registry-modules/actions/workflows/avm.ptn.sa.modernize-your-code.yml/badge.svg?branch=avm%2Fptn%2Fsa%2Fmodernize-yourcode)](https://github.com/Ravikirana-Microsoft/bicep-registry-modules/actions/workflows/avm.ptn.sa.modernize-your-code.yml)

## Type of Change

<!-- Use the checkboxes [x] on the options that are relevant. -->

- Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation
- [ ] Update to CI Environment or utilities (Non-module affecting changes)

## Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] I have run `Set-AVMModule` locally to generate the supporting module files.
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [ ] I have updated the module's CHANGELOG.md file with an entry for the next version

<!--  Please keep up to date with the contribution guide at https://aka.ms/avm/contribute/bicep -->

Kingshuk-Microsoft committed:
AVM-RES: Update virtual network and storage account modules

- Deleted the old virtual network module and replaced it with a new implementation.
- Updated storage account module to use a new SKU default and improved parameter handling.
- Enhanced the test cases for enterprise-grade configurations, including new parameters for Azure AI services.
- Bumped version to 0.2 to reflect the changes made across modules.

Kingshuk-Microsoft committed:
FIX: Corrected wording in VM size notes

- Updated the description for Standard_DS2_v2 to replace "it’s" with "it is" for clarity.

Kingshuk-Microsoft committed:
AVM-PTN: Update workflow paths for modernization

Commit:
Merge branch 'main' into avm/ptn/sa/modernize-yourcode

Commit:
Merge branch 'main' into avm/ptn/sa/modernize-yourcode

Commit:
Merge branch 'main' into avm/ptn/sa/modernize-yourcode

Commit:
fix: regenerated the main.json & readme files

sethsteenken committed:
Merge branch 'main' into avm/ptn/sa/modernize-yourcode

sethsteenken commented on avm/ptn/sa/modernize-your-code/modules/cosmosDb.bicep:
```diff
@@ -5,12 +5,11 @@ param name string
 
 @description('Optional. Tags to be applied to the resources.')
-param tags resourceInput<'Microsoft.Resources/resourceGroups@2025-04-01'>.tags = {}
+param tags object = {}
```
Is there a specific reason these tag types where reverted back to `object`? In the past, we made an effort to update these to target the strict type `resourceInput<'Microsoft.Resources/resourceGroups@2025-04-01'>.tags`.

sethsteenken reviewed (commented):
Looks good, thank you! I added a few comments and questions on some items that need some clarity. Please reach out to me if you have any questions. @Kingshuk-Microsoft @aniaroramsft

Kingshuk-Microsoft committed:
AVM-PTN: Refactor AI Services and Cosmos DB modules

- Removed the ai-services.bicep module as it is no longer needed.
- Updated aifoundry.bicep to remove commented-out code related to cognitive services.
- Changed tags parameter type in cosmosDb.bicep for better resource management.

Commit:
Merge branch 'main' into avm/ptn/sa/modernize-yourcode

Commit:
Merge branch 'main' into avm/ptn/sa/modernize-yourcode

Commit:
Merge branch 'main' into avm/ptn/sa/modernize-yourcode

eriqua reviewed (approved):
🆗

Commit:
Merge branch 'main' into avm/ptn/sa/modernize-yourcode

Commit:
Merge branch 'main' into avm/ptn/sa/modernize-yourcode
